src/app/app.component.ts

```diff
@@ -49,42 +49,71 @@ import { CountdownClock } from './countdownClock';
     <div class='row'>
       <div class='col s12 center-align'>
-        <h1>Elliott &amp; Tracy</h1>
+        <h1 [style.color]="clock.oppositeColor">Elliott &amp; Tracy</h1>
       </div>
     </div>
     <div class="row" id="clockdiv">
-      <div>
-          <span class="days">{{clock.days}}</span>
-          <div class="smalltext">Days</div>
+      <div [style.backgroundColor]="clock.oppositeColor">
+          <span [style.backgroundColor]="clock.color" [style.color]="clock.oppositeColor" class="days">{{clock.days}}</span>
+          <div class="smalltext" [style.color]="clock.color">Days</div>
       </div>
-      <div>
-          <span class="hours">{{clock.hours}}</span>
-          <div class="smalltext">Hours</div>
+      <div [style.backgroundColor]="clock.oppositeColor">
+          <span [style.backgroundColor]="clock.color" [style.color]="clock.oppositeColor" class="hours">{{clock.hours}}</span>
+          <div class="smalltext" [style.color]="clock.color">Hours</div>
       </div>
-      <div>
-          <span class="minutes">{{clock.minutes}}</span>
-          <div class="smalltext">Minutes</div>
+      <div [style.backgroundColor]="clock.oppositeColor">
+          <span [style.backgroundColor]="clock.color" [style.color]="clock.oppositeColor" class="minutes">{{clock.minutes}}</span>
+          <div class="smalltext" [style.color]="clock.color">Minutes</div>
       </div>
-      <div>
-          <span class="seconds">{{clock.seconds}}</span>
-          <div class="smalltext">Seconds</div>
+      <div [style.backgroundColor]="clock.oppositeColor">
+          <span [style.backgroundColor]="clock.color" [style.color]="clock.oppositeColor" class="seconds">{{clock.seconds}}</span>
+          <div class="smalltext" [style.color]="clock.color">Seconds</div>
       </div>
     </div>
   </div>
   `
 })
-export class AppComponent implements OnInit {
-  clock: CountdownClock
+export class AppComponent implements OnInit, OnDestroy {
+  clock = {
+    total: 0,
+    seconds: 0,
+    minutes: 0,
+    hours: 0,
+    days: 0,
+    color: 'black'
+  };
 
-  constructor() {
-    var endDate = new Date(1477152000000);
-    this.clock = new CountdownClock(endDate);
-  }
+  subscription: Subscription;
 
   ngOnInit() {
-    setInterval(() => this.clock.tick(), 1000);
+    const endDate = 1477152000000;
+    const colors = [ 'red', 'orange', 'yellow', 'lime', 'green', 'blue', 'purple' ];
+    const oppositeColors = [ 'lime', 'blue', 'purple', 'red', 'black', 'orange', 'yellow' ];
+    let i = 0;
+    this.subscription = Observable.timer(0, 1000)
+      .map(() => endDate - Date.now())
+      .map(ms => ({
+        total: ms,
+        seconds: Math.floor((ms / 1000) % 60),
+        minutes: Math.floor((ms / 1000 / 60) % 60),
+        hours: Math.floor((ms / (1000 * 60 * 60)) % 24),
+        days: Math.floor(ms / (1000 * 60 * 60 * 24)),
+        color: colors[i++ % colors.length],
+        oppositeColor: oppositeColors[i++ % oppositeColors.length]
```

Comment:
How about this https://github.com/efouts/countdown/commit/4000268d1b302ae6bd175b21c758de9b225b9fc6 ?